client/src/cordova/apple/OutlineAppleLib/Sources/OutlineTunnel/OutlineVpn.swift

```diff
@@ -83,28 +87,39 @@ public class OutlineVpn: NSObject {
 
   // Starts a VPN tunnel as specified in the OutlineTunnel object.
   public func start(_ tunnelId: String, name: String?, configJson: [String: Any], _ completion: @escaping (Callback)) {
```

Comment:
This code was a major pain. Connecting to a server while another server was connected would just not work. That's because the original code wouldn't actually tear down the previous connection. It would replace it instead, and the state tracking was a nightmare. We now cleanly stop the previous connection before starting a new one.

It required a bunch of changes, but now the events work correctly.